app/fissile.go

```diff
@@ -853,15 +853,49 @@ func (f *Fissile) GenerateKube(roleManifestPath, outputDir, repository, registry
 		return err
 	}
 
+	registryCredentials, err := kube.MakeRegistryCredentials(createHelmChart)
+	if err != nil {
+		return err
+	}
+
+	if err = f.generateRegistrySecret(outputDir, registryCredentials, createHelmChart); err != nil {
+		return err
+	}
+
 	if createHelmChart {
-		if err = f.generateHelmValues(outputDir, roleManifest, defaults); err != nil {
+		if err = f.generateHelmValues(outputDir, roleManifest, defaults, registry, username, password, organization); err != nil {
 			return err
 		}
 	}
 
 	return f.generateKubeRoles(outputDir, repository, registry, organization, fissileVersion, roleManifest, defaults, refs, useMemoryLimits, createHelmChart, opinions)
 }
 
+func (f *Fissile) generateRegistrySecret(outputDir string, secret helm.Node, createHelmChart bool) error {
```

Comment:
We now have at least 3 functions that are all just writing out a `helm.Node` to a file, with very little variation, so maybe this can be refactored into a single function?

Reply:
Fixed in 525c1bd